cmd/memtierd/main.go

```diff
@@ -133,16 +170,8 @@ func main() {
 		}
 	}
 
-	for r, routine := range routines {
-		if policy != nil {
-			if err := routine.SetPolicy(policy); err != nil {
-				exit("error in setting policy for routine: %s", err)
-			}
-		}
-		if err := routine.Start(); err != nil {
-			exit("error in starting routine %d: %s", r+1, err)
-		}
-	}
+	setupRoutine(routines, policy)
```

Comment:
As mentionted, let's call this `startRoutines(routines, policy, prompt)`, and include lines 175-177 into it too.

And for the symmetry, you could move lines 164-171 into similar function
`startPolicy(policy, prompt)`.